src/components/Environments/List/EnvironmentListView.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react'
+import EnvEmptyStates from '../EnvEmptyStates'
+import { ReactComponent as EnvIcon } from '../../../assets/icons/ic-environment-list.svg'
+import { NavLink, useHistory, useLocation, useRouteMatch } from 'react-router-dom'
+import { Pagination, Progressing, useAsync } from '../../common'
+import { getEnvAppList } from '../EnvironmentListService'
+import { EnvAppList } from '../EnvironmentGroup.types'
+import { toast } from 'react-toastify'
+
+export default function EnvironmentsListView({ clearSearch }) {
+    const match = useRouteMatch()
+    const location = useLocation()
+    const history = useHistory()
+    const [filteredEnvList, setFilteredEnvList] = useState<EnvAppList[]>([])
+    const [envCount, setEnvCount] = useState<number>()
+    const [paginationParamsChange, setPaginationParamsChange] = useState({ pageSize: 20, offset: 0 })
+    const params = new URLSearchParams(location.search)
+    const paramObj = {
+        envName: params.get('search'),
+        clusterIds: params.get('cluster'),
+        offset: params.get('offset'),
+        size: params.get('pageSize'),
+    }
+    const [paramsData, setParamsData] = useState(paramObj)
+    const [loading, appList] = useAsync(() => getEnvAppList(paramsData), [paramsData])
+    const emptyStateData = paramObj.clusterIds
+        ? { title: 'No app groups found', subTitle: "We couldn't find any matching app groups." }
+        : { title: '', subTitle: '' }
+
+    useEffect(() => {
+        setParamsData(paramObj)
+    }, [location.search])
```

Comment:
Can we directly use `paramObj` instead of setting state inside useEffect

Reply:
Its going in an infinite loop

Reply:
how using the `paramObj` is causing infinite loop?

Reply:
we are calling api on param changes